src/webgpu/api/validation/createTexture.spec.ts

```diff
@@ -70,45 +71,74 @@ class F extends ValidationTest {
 
 export const g = makeTestGroup(F);
 
-g.test('validation_of_sampleCount')
-  .params([
-    // TODO: Consider making a list of "valid"+"invalid" texture descriptors in capability_info.
-    { sampleCount: 0, _success: false }, // sampleCount of 0 is not allowed
-    { sampleCount: 1, _success: true }, // sampleCount of 1 is allowed
-    { sampleCount: 2, _success: false }, // sampleCount of 2 is not allowed
-    { sampleCount: 3, _success: false }, // sampleCount of 3 is not allowed
-    { sampleCount: 4, _success: true }, // sampleCount of 4 is allowed
-    { sampleCount: 8, _success: false }, // sampleCount of 8 is not allowed
-    { sampleCount: 16, _success: false }, // sampleCount of 16 is not allowed
-    { sampleCount: 4, mipLevelCount: 2, _success: false }, // multisampled multi-level not allowed
-    { sampleCount: 4, arrayLayerCount: 2, _success: false }, // multisampled multi-layer is not allowed
-  ])
+g.test('mipLevelCount_validation_on_formats')
+  .params(poptions('format', kAllTextureFormats))
```

Comment:
All of these are expected to pass, which means there's no control case or boundary condition being tested. How about something like this (using subcases from #444):

```ts
g.test('mipLevelCount_bound,formats')
  .desc(`Test mipLevelCount=6 at the boundary between size 31 and 32,
  with every dimension, and every possible format.`)
  .cases(params()
    .combine(poptions('dimension', ['1d', '2d', '3d']))
    .combine(poptions('format', kAllTextureFormats))
  ).subcases(({ dimension }) =>
    dimension === '1d' ? [
      { ok: true, size: [32, 1] },
      { ok: false, size: [31, 1] },
    ] : dimension === '2d' ? [
      { ok: true, size: [32, 31, 1] },
      { ok: true, size: [31, 32, 1] },
      { ok: false, size: [31, 31, 1] },
    ] : [
      { ok: true, size: [32, 31, 31] },
      { ok: true, size: [31, 32, 31] },
      { ok: true, size: [31, 31, 32] },
      { ok: false, size: [31, 31, 31] },
    ])
```

This tests different dimensions, but some dimension+format combinations might not be valid, so we'd have to filter those out (e.g. I don't think we allow 3d compressed textures yet). We can skip testing multiple dimensions, but if we do, but be sure to leave a TODO in the `.desc()`.

Perhaps kAllTextureFormats should be replaced by a hand-picked representative list (say, a few regular formats, all depth/stencil if valid\*, a few compressed).

This does end up overlapping a bit with the next test; I think a few cases from that can just be removed if they're covered here.

\* Not sure if depth/stencil mipmapping is allowed. We should check into this, perhaps first by checking against Dawn and then Vulkan. If you don't find a conclusive answer, we should err on the side of testing too many formats.

Reply:
Thanks for your suggestion, Kai. I think the test mipLevelCount,bound_check (the following one) intend to test the boundary of mipLevelCount (how to calculate the max value of mipLevelCount correctly) according to texture size/dimension. So I think we don't need to add a lot of boundary check for mipLevelCount. However, the `subcases` is interesting. I will try to use it if needed.

Reply:
I suppose this is resolved.